ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/infrastructure/LedgerTestSuite.scala

```diff
@@ -32,6 +32,33 @@ private[testtool] abstract class LedgerTestSuite {
       repeated,
     )((ec: ExecutionContext) => (_: Seq[ParticipantTestContext]) => testCase(ec))
   }
+
+  // TODO PBATKO: How to name it? Differs from test() method in the type `testCase` parameter (here `testCase` function accepts also `Seq[ParticipantTestContext]`)
+  protected final def test2(
+      shortIdentifier: String,
+      description: String,
+      participants: ParticipantAllocation,
+      timeoutScale: Double = 1.0,
+      runConcurrently: Boolean = true,
+      repeated: Int = 1,
+  )(
+      testCase: ExecutionContext => Seq[ParticipantTestContext] => PartialFunction[
+        Participants,
+        Future[Unit],
+      ]
+  ): Unit = {
+    testGivenAllParticipants(
+      shortIdentifier,
+      description,
+      participants,
+      timeoutScale,
+      runConcurrently,
+      repeated,
+    )((ec: ExecutionContext) =>
+      (testContext: Seq[ParticipantTestContext]) => testCase(ec)(testContext)
```

Comment:
TLS1.3's `testCase` needs `testContext` to inspect it for participant hostname and port which are then used to construct a collection of gRPC channels each exercising different TLS version on the client side.

This seems like an abuse of test tool's framework where `testCase`'s is already effectively a client over some/none TLS.
For the need of our new TLSv1.3 test case we effectively ignore that pre-made client. (We only care for it's `TlsConfiguration` to be able to create our own with only TLS version modified)

Reply:
Maybe rather than wrapping it in test2, just use testGivenAllParticipants directly in TLSOnePointThreeIT. There is prior art with e.g. KVCommandDeduplicationBase